### ISSUE TITLE
Overriding public_id method in carrierwave uploader causes update to only delete existing image

### ISSUE THREAD

**Issue by jpawlyn** [cloudinary/cloudinary_gem#267]
Carrierwave gem - 1.1.0
Cloudinary gem -  1.8.1

When the `public_id` method is overridden in an uploader, image creation works fine but if the image is subsequently updated, the existing image is deleted and no new image is uploaded although a new url reference is written to the db. I could be doing something wrong so if it's working fine for others that would be good to know. I've tried downgrading to carrierwave 1.0.0 but with the same result. Thanks,

Jolyon

**Comment by roeeba** [comment]
Hi, @jpawlyn. Can you please share with us your Ruby/Rails versions, so we can try and reproduce it on our side as well? Thanks!

**Comment by jpawlyn** [comment]
Yup, sorry for not including that. It is Rails 5.0.5 and Ruby 2.3.3.

**Comment by roeeba** [comment]
Hi, @jpawlyn. Did you try to replace the image with a different one, or remove it completely?
We've tried to replace images and it seems to work for us here with (almost) the same versions as you use. (Ruby 2.3.3, Rails 5.1.4, Cloudinary 1.8.1, CW 1.1.0).

Here's the project we run:
https://github.com/taragano/Carrierwave-override-public-id

**Comment by jpawlyn** [comment]
Hi, sorry for the delay in responding @roeeba. Removing the image worked fine, but replacing it failed as described. I've looked at the example project and the only significant difference I noticed between the `photo_uploader.rb` and the uploader we are using is that we override the `store_dir` method. Could that account for the issue do you think? Thanks.

**Comment by roeeba** [comment]
Hi @jpawlyn. We apologize for not responding earlier. Not sure it's relevant anymore, but here are our insights:
If you need to upload a new image, while keeping the old one, then you'll need to set up your rails app differently.
If you need your app to have both old and new images, then you'll need your app to be able to add more images rather than having a single updated one, and also make sure every new upload will assign a unique public ID to every upload.
On the other hand, if you're ok with your app having only a single (latest) version of the image, but still want to not delete the old versions on your Cloudinary account, then you'll need to also make sure the public IDs are unique for each new upload.
